File: src/Budget/Budget.Core/Models/Records/UpdateRecordModel.cs
using Budget.Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Budget.Core.Models.Records
{
    public class UpdateRecordModel
    {
        public int Id { get; set; }

        public string Note { get; set; }

        public decimal Amount { get; set; }

        public int AccountId { get; set; }

        public int CategoryId { get; set; }

        public int PaymentTypeId { get; set; }

        public RecordType RecordType { get; set; }
    }
}

File: web/budget-web/src/app/shared/models/records/update-record.model.ts
export class UpdateRecordModel {
  constructor(
    public id: number,
    public note: string,
    public amount: number,
    public accountId: number,
    public categoryId: number,
    public paymentTypeId: number,
    public recordType: string,
    public recordDate: Date,
  ) {}
}
